android/src/main/java/org/reactnative/camera/tasks/BarCodeScannerAsyncTask.java

```diff
@@ -41,14 +41,27 @@ protected Result doInBackground(Void... ignored) {
       BinaryBitmap bitmap = generateBitmapFromImageData(mImageData, mWidth, mHeight);
       result = mMultiFormatReader.decodeWithState(bitmap);
     } catch (NotFoundException e) {
-      // No barcode found, result is already null.
+      BinaryBitmap bitmap = generateBitmapFromImageData(rotateImage(mImageData,mWidth, mHeight),mHeight,mWidth);
+      try {
+        result = mMultiFormatReader.decodeWithState(bitmap);
+      } catch (NotFoundException e1) {
+        //no barcode Found
+      }
     } catch (Throwable t) {
       t.printStackTrace();
     }
 
     return result;
   }
-
+  private byte[] rotateImage(byte[]imageData,int width, int height) {
+    byte[] rotated = new byte[imageData.length];
+    for (int y = 0; y < height; y++) {
+      for (int x = 0; x < width; x++) {
+        rotated[x * height + height - y - 1] = imageData[x + y * width];
```

Comment:
this may lead to crashes, that is why there was an `if` inside the `for` on the old method.

Reply:
Please separate into `sourceIx` and `destIx` and check if those are valid, like before

Reply:
Ok